app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -77,8 +66,9 @@ def destroy
 
   private
 
-  def find_bookmark
+  def bookmark_ownership_required
     return unless (@bookmark = find_object(Bookmark))
-    access_denied unless @bookmark.visible_to?(current_user)
+    access_denied and return unless @bookmark.visible_to?(current_user)
+    access_denied unless @bookmark.user.id == current_user.try(:id)
```

Comment:
```suggestion
    access_denied unless @bookmark.user.id == current_user.try(:id)
```
This should be enough, right?

Reply:
No bc a post that has a replied bookmark may have become private since the bookmark was added and so the user can't access the bookmark even if they're the owner

Reply:
discussed aloud, kinda awkward but seems legit